### PR TITLE
implemented phases in TurnTracker

### PR DESCRIPTION
**Pull request by danny-does-stuff**
TurnTracker now keeps track of phases during a turn as well as setup turn.

### CODE REVIEW

File: java/src/shared/model/game/TurnTracker.java [hunk 1]
```diff
@@ -6,8 +6,11 @@
  * Representation of Player Turns
  */
 public class TurnTracker {
+    private static final int NUMBER_OF_PHASES = 3;
+
+    int setupTurn;
     private int currentTurn;
-    private String status;
+    private String status; //'Rolling' or 'Robbing' or 'Playing' or 'Discarding' or 'FirstRound' or 'SecondRound'
```
Comment (CaptBoscho) [inline java/src/shared/model/game/TurnTracker.java]: make the status an int and rotate through for the status.

File: java/src/shared/model/game/TurnTracker.java [hunk 2]
```diff
@@ -61,11 +68,41 @@ public int getCurrentTurn() {
...
+            } else {
+                throw new Exception("Current setup turn is invalid.  This is broken.");
+            }
+        } else if (status.equals("SecondRound")) {
+            return setupTurn--;
+        } else {
+            throw new Exception("Invalid status. Can only do setup when status is 'FirstRound' or 'SecondRound'");
+        }
+    }
+
```
Comment (CaptBoscho) [inline java/src/shared/model/game/TurnTracker.java]: perhaps make canRoll, canTrade, canBuild which checks the if the status is the number (phase) we want.
Then an incrementStatus function

File: java/src/shared/model/game/TurnTracker.java [hunk 3]
```diff
@@ -61,11 +68,41 @@ public int getCurrentTurn() {
...
 
     /**
+     * Goes to the next players turn during the setup phase.
+     * @return The index of the next player to setup
+     * @throws Exception if setupTurn > getNumPlayers()
+     */
+    public int nextSetupTurn() throws Exception {
+        if (status.equals("FirstRound")) {
+            if (setupTurn < getNumPlayers()) {
+                return setupTurn++;
```
Comment (CaptBoscho) [inline java/src/shared/model/game/TurnTracker.java]: what about the player with index of 0?

File: java/src/shared/model/game/TurnTracker.java [hunk 4]
```diff
@@ -61,11 +68,41 @@ public int getCurrentTurn() {
     }
 
     /**
+     * Goes to the next players turn during the setup phase.
```
Comment (CaptBoscho) [inline java/src/shared/model/game/TurnTracker.java]: could we also get a function checking if it the players turn during the setup round? so passes in a playerID, and checks if it's that's player's turn

File: java/src/shared/model/game/TurnTracker.java (hunk 1 above)
Comment (derekargueta) [inline java/src/shared/model/game/TurnTracker.java]: Better yet, use an enum

File: java/src/shared/model/game/TurnTracker.java (hunk 4 above)
Comment (dannyharding10) [inline java/src/shared/model/game/TurnTracker.java]: done

**Pull request merged**